cranelift/interpreter/src/interpreter.rs

```diff
@@ -191,9 +192,13 @@ pub enum HeapInit {
     FromBacking(HeapBacking),
 }
 
+pub type LibCallHandler<'a, V> =
+    &'a dyn Fn(LibCall, SmallVec<[V; 1]>) -> Result<SmallVec<[V; 1]>, TrapCode>;
```

Comment:
Since this `SmallVec` type ends up appearing in other modules, I'd suggest introducing a type alias for it. That way it's easy later either to change it to a regular `Vec`, or to put more elements in the inline array.

Also, `&dyn Fn` is probably more general than necessary. A closure that doesn't capture anything from its environment is compatible with a `fn` type, and I'd be surprised if interpreter libcall handlers ever needed context.

```suggestion
pub type LibCallValues<V> = SmallVec<[V; 1]>;
pub type LibCallHandler<V> = fn(LibCall, LibCallValues<V>) -> Result<LibCallValues<V>, TrapCode>;
```